server/server.go

```diff
@@ -283,14 +283,20 @@ func (s *Server) WithDecisionIDFactory(f func() string) *Server {
 }
 
 // Listeners returns functions that listen and serve connections.
-func (s *Server) Listeners() (func() error, func() error, error) {
+func (s *Server) Listeners() ([]func() error, error) {
 	parsedURL, err := url.Parse(s.addr)
 
 	// url.Parse won't necessarily return an error if there isn't a scheme
 	if err == nil && parsedURL.Scheme != "" {
-		return s.getListenerFromURL(parsedURL)
+		loops, err := s.getListenerFromURL(parsedURL)
+		if err != nil {
+			return nil, err
+		}
+		return loops, nil
```

Comment:
You are right. I was developing this iteratively (just change that function first. And instead of reverting back, this stayed. Gotta change it, thanks!